### PR TITLE
Pr wind est rebased

### PR DESCRIPTION
**Pull request by RomanBapst**
Rebased replacement of #8273

### CODE REVIEW

**Comment by RomanBapst** [comment]
@dagar I forgot to pull your changes on the old branch so I'm missing your work on having it publish to a second instance of the wind topic. I will pull that stuff over again.

@sanderux We are going to test fly this today on a system with and without airspeed sensor.

**Comment by RomanBapst** [comment]
Successful test flight using only sideslip fusion:

Convergence tiltrotor:
https://logs.px4.io/plot_app?log=0b15468e-eaf6-44b4-b45f-746982bd125b

DeltaQuad
https://logs.px4.io/plot_app?log=1e96772b-4e51-4d17-a07f-5255b0d28e6a

**Comment by RomanBapst** [comment]
@PX4/testflights Could you please give this a try on a vehicle capable of flying in fixed wing mode and having an airspeed sensor installed?
I already tested this on a VTOL without airspeed sensor and it works.

**Comment by RomanBapst** [comment]
@bkueng Could do review this, especially regarding memory optimizations.

**Comment by r0gelion** [comment]
@RomanBapst we can test it in a fixed wing with airspeed installed, we might not able to test it until Monday.

**Comment by dagar** [comment]
What are your thoughts on running this in a work queue at a much lower rate? The downside is there will likely be some variability in dt.

**Comment by RomanBapst** [comment]
@dagar I'm ok with that. Variability of dt is ok I think, it does not need to be a constant. As long as we know the value it's fine.

**Comment by RomanBapst** [comment]
@r0gelion No problem, enjoy your weekend!

File: src/modules/sensors/sensors.cpp [hunk 1]
```diff
@@ -346,24 +380,78 @@ Sensors::diff_pres_poll(struct sensor_combined_s &raw)
...
+						   _voted_sensors_update.baro_pressure(), air_temperature_celsius));
+
+		_airspeed.air_temperature_celsius = air_temperature_celsius;
+		_airspeed.differential_pressure_filtered_pa = _diff_pres.differential_pressure_filtered_pa;
+	}
+
+	// update wind and airspeed estimator
+	_wind_estimator.update(hrt_absolute_time());
+
+	bool fuse_airspeed = updated && (hrt_elapsed_time(&_time_last_airspeed_fused) > 5e4);
```
Comment (bkueng) [inline src/modules/sensors/sensors.cpp]: There are many `hrt_absolute_time` calls in this block here

File: src/modules/sensors/sensors.cpp [hunk 2]
```diff
@@ -178,10 +185,19 @@ class Sensors : public ModuleBase<Sensors>
 	orb_advert_t	_diff_pres_pub{nullptr};			/**< differential_pressure */
 	orb_advert_t	_sensor_preflight{nullptr};		/**< sensor preflight topic */
 
+	orb_advert_t 	_wind_est_pub{nullptr};			/**< wind estimate topic */
+
 	perf_counter_t	_loop_perf;			/**< loop performance counter */
 
 	DataValidator	_airspeed_validator;		/**< data validator to monitor airspeed */
 
+	differential_pressure_s _diff_pres{};
```
Comment (bkueng) [inline src/modules/sensors/sensors.cpp]: All of these together with `WindEstimator` add several 100 bytes to the RAM usage, for something that we don't need in MC (or will we?). Something optional & dynamically allocated might make more sense here.

File: src/lib/wind_estimator/WindEstimator.hpp [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+/**
+ * @file WindEstimator.hpp
+ * A wind and airspeed scale estimator.
+ */
+
+#pragma once
+
+#include <mathlib/mathlib.h>
+
+using namespace matrix;
```
[the remark this answers is not in the document]
Comment (RomanBapst) [reply src/lib/wind_estimator/WindEstimator.hpp]: @bkueng Is it legitimate to put
#define Vec2f matrix::Vector2f
#define Vec3f matrix::Vector3f
...
into the header of the wind estimator?

**Comment by santiago3dr** [comment]
Couple flights on a Phantom Flying Wing running a Pixhawk Mini (V3):
https://logs.px4.io/plot_app?log=d6ec2e91-66bb-4324-a2f6-9872fc956613
https://logs.px4.io/plot_app?log=31d1a655-076f-4bb0-adaa-65f31a3c1c7a

Nice flights, no noticeable difference from current master:
https://logs.px4.io/plot_app?log=6c43b832-e8a8-43f5-bd27-7f00f9d78e01

File: src/modules/sensors/sensors.cpp (hunk 2 above)
Comment (sanderux) [reply src/modules/sensors/sensors.cpp]: wind data can be very useful for certain types of sensor readings such as gas sensors

**Comment by RomanBapst** [comment]
@santiago3dr @r0gelion Thanks for testing this! It has actually unveiled a bug in the estimator which was only visible when flying in strong winds. I was able to replicate the error in SITL and fixed it.
I'm going to close this because we are using a different implementation in terms of architecture in #9079 
I'll let you know if it's ready for testing, hopefully you can do it on another windy day :-)

**Comment by santiago3dr** [comment]
sounds good; windy conditions are easy to come by, let us know when the new code is ready for testing

**Pull request closed without merging**